geoalchemy2/types.py

```diff
@@ -147,6 +154,11 @@ def process(bindvalue):
                     return '%s' % (bindvalue.data)
                 else:
                     return 'SRID=%d;%s' % (bindvalue.srid, bindvalue.data)
+            elif isinstance(bindvalue, WKBElement):
+                if not SHAPELY:
+                    raise ArgumentError('Shapely required for handling WKBElement bind values')
+                shape = to_shape(bindvalue)
+                return 'SRID=%d;%s' % (bindvalue.srid, shape.wkt)
```

Comment:
I think that converting WKB back to WKT is quite disappointing (as far as I observed, WKT can sometimes loose precision and are slower to insert) but I don't know a clean method to use only WKB at this moment. But as soon as `shapely==1.7` is released, I think that this line should be `return wkb.dumps(shape, hex=True, srid=bindvalue.srid)` (see the new `wkb.dumps`: https://github.com/Toblerity/Shapely/blob/master/shapely/wkb.py#L24). Note that this is possible because `ST_GeomFromEWKT()` actually accepts either WKT, EWKT, WKB or EWKB argument value (I only tested with postgis, I don't know how sptialite behaves).

Reply:
Good call. I'll try a bit harder to see if we can avoid the wkt conversion.

I didn't know `ST_GeomFromEWKT` worked with WKB hex strings. This is surprising.

Reply:
Maybe we can merge my code as-is and revisit it when Shapely 1.7 is available. Is that what you're suggesting Adrien ?

Reply:
Yes since I don't know how to do better than this fix without the next release of shapely (except by copying the new code of shapely here, which is event more disappointing) and it is much better than a failure anyway :-)

Reply:
@adrien-berchet I checked, and SpatiaLite's `GeomFromEWKT` does not support (E)WKB hex strings, as expected.

But please have a look at my last commit. There is a case where we can avoid the EWKB to EWKB conversion.

Thank you for letting me know about PostGIS `ST_GeomFromEWKT` being able to work with (E)WKB hex strings.

Reply:
This is nice since the `WKBElement`s returned by queries will be extended, so we should almost always use the `return bindvalue.desc` case when working with PostGIS, except when the `WKBElement`s are explicitely created with `extended=False`, which seems unlikely in real use case.

Note: a test for extended `WKBElement` should be added.

Reply:
> This is nice since the WKBElements returned by queries will be extended, so we should almost always use the return bindvalue.desc case when working with PostGIS, except when the WKBElements are explicitely created with extended=False, which seems unlikely in real use case.

Exactly. That was my thought as well.

> Note: a test for extended WKBElement should be added.

What do you mean? Are you talking about a test related to this PR's additions? Or something else?

Reply:
> What do you mean? Are you talking about a test related to this PR's additions? Or something else?

A test related to this PR. The same as `test_select_bindparam_WKBElement` but with an extended `WKBElement`. Because the `return bindvalue.desc` case seems to be not covered.

Reply:
Ok, will have a look this afternoon.

Reply:
New test added. I'll merge if Travis is green.

Reply:
And I'll create a ticket suggesting the use of `wkb.dumps(srid=xxx)` when Shapely 1.7 is available.